src/Storefront/Resources/app/storefront/src/plugin/forms/form-handler.plugin.js

```diff
@@ -0,0 +1,233 @@
+import Plugin from 'src/plugin-system/plugin.class';
+import Debouncer from 'src/helper/debouncer.helper';
+import ButtonLoadingIndicator from 'src/utility/loading-indicator/button-loading-indicator.util';
+
+/**
+ * @module FormHandler - Form Handler Plugin
+ *
+ * This plugin is used to apply form handling to an HTML form.
+ * It's main features are best-practices for accessible form validation.
+ * The validation makes use of the global `FormValidation` class.
+ *
+ * You can use the Twig form templates in `storefront/component/form` to get
+ * ready to use form fields that implement all the best-practices.
+ *
+ * @example
+ * <form data-form-handler="true">
+ *
+ *     {% sw_include '@Storefront/storefront/component/form/form-input.html.twig' with {
+ *         label: 'account.personalFirstNameLabel'|trans|sw_sanitize,
+ *         id: 'personalFirstName',
+ *         name: 'firstName',
+ *         value: data.get('firstName'),
+ *         autocomplete: 'section-personal given-name',
+ *         violationPath: violationPath,
+ *         validationRules: 'required',
+ *         additionalClass: 'col-sm-6',
+ *     } %}
+ *
+ *     <button type="submit">Submit</button>
+ * </form>
+ */
+export default class FormHandler extends Plugin {
+
+    static options = {
+
+        /**
+         * Define if the form should use form validation.
+         *
+         * @type {boolean}
+         */
+        validation: true,
+
+        /**
+         * Define if the form should be validated before submit.
+         *
+         * @type {boolean}
+         */
+        validateOnSubmit: true,
+
+        /**
+         * Define if the first invalid field should be focused,
+         * if the form is validated on submit.
+         *
+         * @type {boolean}
+         */
+        focusInvalidField: true,
+
+        /**
+         * Define the debounce time of the field validation.
+         *
+         * @type {string}
+         */
+        debounceTime: '200',
+
+        /**
+         * Define the selector for gathering all fields of the form.
+         *
+         * @type {string}
+         */
+        formFieldSelector: 'input, textarea, select',
+
+        /**
+         * Define if a loading indicator should be shown on the submit button.
+         *
+         * @type {boolean}
+         */
+        loadingIndicator: true,
+
+        /**
+         * @type {'before' | 'after' | 'inner'} loadingIndicatorPosition
+         */
+        loadingIndicatorPosition: 'before',
+    };
+
+    init() {
+        if (this._isFormElement(this.el) === false) {
+            throw Error('Element is not of type <form>');
```

Comment:
```suggestion
            throw new Error('Element is not of type <form>');
```
Is new missing here?

Reply:
Seems to be a copy-paste thing. I changed it to console error because I'm not a big fan of hard errors on the client side anyway.

Reply:
I would also prefer this or `console.error` over a hard error. 👍